client/client.go

```diff
@@ -148,8 +148,15 @@ func (c *Client) init() error {
 			return fmt.Errorf("failed creating state dir: %s", err)
 		}
 
-		c.logger.Printf("[INFO] client: using state directory %v", c.config.StateDir)
+	} else {
```

Comment:
I think that is a separate question to be addressed in a different PR. Right now the system assumes you have a state_dir: https://github.com/hashicorp/nomad/blob/master/client/client.go#L319.

A different PR could disable state saving.